src/App.jsx

```diff
@@ -14,33 +16,145 @@
+          break;
+        case SORT_BY_LENGTH:
+          newGoods = [...sortedGoods].sort(
+            (good1, good2) => good2.length - good1.length,
+          );
+          setAlphPressed(false);
+
+          break;
+        case REVERSE:
+          newGoods = [...sortedGoods].reverse();
+          break;
+        case RESET:
+          newGoods = [...goodsFromServer];
+          setReversePressed(false);
+          setAlphPressed(false);
+          setLengthPressed(false);
+          break;
+        default:
+          break;
+      }
+    } else {
+      switch (type) {
+        case SORT_ALPHABETICALLY:
+          newGoods.sort((good1, good2) => good1.localeCompare(good2));
+          setLengthPressed(false);
+          break;
+        case SORT_BY_LENGTH:
+          newGoods.sort((good1, good2) => good1.length - good2.length);
+          setAlphPressed(false);
+
+          break;
+        case REVERSE:
+          newGoods = [...sortedGoods].reverse();
+          break;
+        case RESET:
+          newGoods = [...goodsFromServer];
+          setReversePressed(false);
+          setAlphPressed(false);
+          setLengthPressed(false);
+          break;
+        default:
+          break;
+      }
+    }
+
+    setSortedGoods(newGoods);
+    setSort(type);
+  };
 
-      <button type="button" className="button is-danger is-light">
+  const renderResetButton = () => {
+    return (
+      <button
+        type="button"
+        className="button is-danger"
+        onClick={() => {
+          handleSort(RESET);
+        }}
+      >
         Reset
       </button>
-    </div>
+    );
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': alphPressed !== true,
+            "NOT_ACTIVE_CLASS": sort !== SORT_ALPHABETICALLY,
+          })}
+          onClick={() => {
+            handleSort(SORT_ALPHABETICALLY);
+            setAlphPressed(!alphPressed);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            handleSort(SORT_BY_LENGTH);
+            setLengthPressed(!lengthPressed);
+          }}
+          className={classNames('button is-success', {
+            'is-light': lengthPressed !== true,
+            "NOT_ACTIVE_CLASS" : sort !== SORT_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': reversePressed !== true,
+            "NOT_ACTIVE_CLASS": sort !== REVERSE,
```

Comment:
The class `NOT_ACTIVE_CLASS` is used here but is not defined or imported. This will cause an error. Please ensure that this class is defined or imported correctly.